tests/src/test/scala/actionContainers/ActionContainer.scala

```diff
@@ -65,13 +67,26 @@ trait ActionProxyContainerTestUtils extends FlatSpec with Matchers with StreamLo
   def checkStreams(out: String,
                    err: String,
                    additionalCheck: (String, String) => Unit,
-                   sentinelCount: Int = 1): Unit = {
+                   sentinelCount: Int = 1,
+                   concurrent: Boolean = false): Unit = {
     withClue("expected number of stdout sentinels") {
       sentinelCount shouldBe StringUtils.countMatches(out, sentinel)
     }
+    //sentinels should be all together
+    if (concurrent) {
```

Comment:
this seems overly strict - it will mean the runtimes have to guarantee not to interleave sentinels as activations finish; is that the intent?

Reply:
could you just count the number of sentinel occurrences to match the number of activations?

Reply:
That is done already, but what is important is the unknown ordering of sentinels vs log output since these should be different when container runs activations concurrently (without concurrency, the container logs always alternate between <log><sentinel><log><sentinel>...). With concurrent activations, the sentinel from previous activation may output in between logs of subsequent activations, like:
```
<activation1 log>
<activation1 log>
<activation2 log>
<activation1 sentinel>
<activation2 log>
<activation2 sentinel>
```

So in the nodejs container test I will force the activation logs to all be output before any sentinels are output.

Reply:
did you remove this change or did I botch the commits?

Reply:
I removed it from this base class, moved it to https://github.com/apache/incubator-openwhisk-runtime-nodejs/pull/41/commits/07303469b3ac681903a6307a9479875524c195a9#diff-a3608ee6dd1352899130c2c7a2612301R595

Reply:
Phew. Thanks.